File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int n = nums.length;
        int s = 0;
        int e  = n-1 ;
        while(s<=e){
            int mid = s +(e-s)/2;
            System.out.println(mid);
            if(nums[mid]==target) return mid;
            else if(nums[mid]>target){
                e = mid-1;
            }else{
                s = mid+1;
            }
        }
        return -1;
    }
}

File: 0704-binary-search/0704-binary-search.ts
function search(nums: number[], target: number): number {
  let low: number =0;
    let high: number =nums.length-1;
    while(low<=high){
        const mid: number = Math.floor( (low+(high-low)/2));
        console.log(mid);
        if(nums[mid]==target) return mid;
        else if(nums[mid]>target) high = mid-1;
        else low = mid+1;
    }
    return -1;
};